### ISSUE TITLE
[libxml2] Build error on x64-windows

### ISSUE THREAD

**Issue by sergeikobelev** [microsoft/vcpkg#41754]
Package: libxml2[core,iconv,lzma,zlib]:x64-windows@2.11.9

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.40.33813.0
-    vcpkg-tool version: 2024-10-18-e392d7347fe72dff56e7857f7571c22301237ae6
    vcpkg-scripts version: 41626fd77 2024-10-23 (11 hours ago)

**To Reproduce**

`vcpkg install libxml2`

**Failure logs**

```
-- Downloading https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz -> GNOME-libxml2-v2.11.9.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from D:\vcpg\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: D:\vcpg\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
error: Missing GNOME-libxml2-v2.11.9.tar.gz and downloads are blocked by x-block-origin.
error: https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz: failed: status code 500
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030): 
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 3us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 996us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 11 s (10592837us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_gitlab.cmake:113 (vcpkg_download_distfile)
  ports/libxml2/portfile.cmake:1 (vcpkg_from_gitlab)
  scripts/ports.cmake:192 (include)



```

**Comment by sergeikobelev** [comment]
Most likely sources don't available anymore on specified site.

**Comment by WangWeiLin-MV** [comment]
Not repro, the URL is available.

**Comment by Hind-M** [comment]
Hey there,
I have the same problem as well. Is there a fix or a workaround for this?
Thanks!

**Comment by WangWeiLin-MV** [comment]
The link in the issue still available https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz

**Comment by Hind-M** [comment]
Oh right, it doesn't seem to fail anymore, thanks!

**Comment by R3D9477** [comment]
`Installing 1/14 libxml2[core,iconv,lzma,zlib]:x64-linux@2.11.9...
Building libxml2[core,iconv,lzma,zlib]:x64-linux@2.11.9...
Downloading GNOME-libxml2-v2.11.9.tar.gz
Failed to download GNOME-libxml2-v2.11.9.tar.gz.
If you are using a proxy, please ensure your proxy settings are correct.
Possible causes are:
1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
This is not correct, because `https://` prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
Try setting `http://address:port` to both HTTP_PROXY and HTTPS_PROXY instead.
2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See, https://github.com/microsoft/vcpkg-tool/pull/77
The value set by your proxy might be wrong, or have same `https://` prefix issue.
3. Your proxy's remote server is our of service.
If you've tried directly download the link, and believe this is not a temporay download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
to report this upstream download server failure.
error: https://gitlab.gnome.org//GNOME/libxml2/-/archive/v2.11.9/libxml2-v2.11.9.tar.gz: curl failed to download with exit code 22
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

curl: (22) The requested URL returned error: 503

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:231 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_gitlab.cmake:113 (vcpkg_download_distfile)
  ports/libxml2/portfile.cmake:1 (vcpkg_from_gitlab)
  scripts/ports.cmake:196 (include)


error: building libxml2:x64-linux failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle libxml2:x64-linux: 5.2 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libxml2
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[libxml2]+Build+error+on+x64-linux&body=Copy+issue+body+from+%2Fhome%2Fbuild%2F.vcpkg%2Fvcpkg_install%2Fvcpkg%2Fissue_body.md`

Is it possible to change URL from unstable original gnome.gitlab to more stable read-only mirror located on github ?

**Comment by remcozut** [comment]
Open /path/to/vcpkg/ports/libxml2/portfile.cmake 

Replace 

vcpkg_from_gitlab(
    GITLAB_URL https://gitlab.gnome.org/
    OUT_SOURCE_PATH SOURCE_PATH
    REPO GNOME/libxml2
    REF "v${VERSION}"
    SHA512 3f2de446657bf3c23c92358ce8946f59253b9fcc09577b59eecaffdbd97e051659855c79f4882ee9f8841dd194b6bd5de2a8017691473b505e905b9dde6a1bc9
    HEAD_REF master
    PATCHES
        disable-docs.patch
        fix_cmakelist.patch
)

with

vcpkg_from_github(
	OUT_SOURCE_PATH SOURCE_PATH
	REPO GNOME/libxml2
	REF "v${VERSION}"
	SHA512 10585f3a9274af6eb6980b0159ea8a82b3f55662c94eac0a43acab87e7feb625f1629d20e0d477ab06826170f9b5b0d817eb8fc8a472afa6001851e7dbf199f3
	HEAD_REF master
	PATCHES
		disable-docs.patch
		fix_cmakelist.patch
)
